src/Behat/Behat/Output/Node/Printer/JUnit/JUnitSetupPrinter.php

```diff
@@ -52,23 +54,20 @@ public function printTeardown(Formatter $formatter, Teardown $teardown)
         }
     }
 
-    /**
-     * @param Formatter $formatter
-     * @param CallResults $results
-     * @param string $messageType
-     */
-    private function handleHookCalls(Formatter $formatter, CallResults $results, $messageType)
+    private function handleHookCalls(Formatter $formatter, CallResults $results, string $messageType): void
     {
-        /** @var CallResult $hookCallResult */
         foreach ($results as $hookCallResult) {
+            assert($hookCallResult instanceof CallResult);
```

Comment:
Prefer to use assertions to @var annotations

Reply:
The right solution here is to add `@template-implements \IteratorAggregate<array_key, CallResult>` on the `CallResults` class, removing the need for local type refinements.

I agree that assertions are better than `@var` because wrong assertions will make our CI fail (assuming we configured GHA to use the development ini config in setup-php), but proper type inference is even better from a type safety point of view, as it is better to fail during type checking than at runtime.

Reply:
The `CallResults` class already had the `@implements` annotation so this assert was not actually needed (and neither a couple of other asserts). I had converted the @var annotations to asserts without checking if they were actually needed, now they have been removed